src/Entity/Form/EdgeContentEntityForm.php

```
@@ -0,0 +1,323 @@
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Entity form variant for Apigee Edge entity types.
+ *
+ * The content `ContentEntityForm` makes the assumption that entities are
+ * translatable and revisionable. This entity base form provides most of the
+ * same helpers for Apigee Edge entities.
+ *
+ * @see \Drupal\Core\Entity\ContentEntityForm
+ */
+class EdgeContentEntityForm extends EntityForm implements EdgeContentEntityFormInterface {
+
+  /**
+   * The entity being used by this form.
+   *
+   * @var \Drupal\apigee_edge\Entity\EdgeEntityInterface
+   */
+  protected $entity;
+
+  /**
+   * The entity type bundle info service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeBundleInfoInterface
+   */
+  protected $entityTypeBundleInfo;
+
+  /**
+   * The time service.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $time;
+
+  /**
+   * The entity repository service.
+   *
+   * @var \Drupal\Core\Entity\EntityRepositoryInterface
+   */
+  protected $entityRepository;
+
+  /**
+   * Constructs a EdgeContentEntityForm object.
+   *
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entity_repository
+   *   The entity repository service.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The entity type bundle service.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   */
+  public function __construct(EntityRepositoryInterface $entity_repository, EntityTypeBundleInfoInterface $entity_type_bundle_info = NULL, TimeInterface $time = NULL) {
+    if ($entity_repository instanceof EntityManagerInterface) {
+      @trigger_error('Passing the entity.manager service to ContentEntityForm::__construct() is deprecated in Drupal 8.6.0 and will be removed before Drupal 9.0.0. Pass the entity.repository service instead. See https://www.drupal.org/node/2549139.', E_USER_DEPRECATED);
```

If this is deprecated in Drupal 8.6.0 and this module is not yet available, should we instead throw an error instead of handling it gracefully to keep developers from integrating w/us the deprecated way?